### ISSUE TITLE
Advanced Filters for routing rules and hooks

### ISSUE THREAD

**Issue by mcweba** [swisspush/gateleen#265]
The current implementations of routing rules, hooks and delegates provide functionality to filter requests based on the URL.

A powerful improvement would be to have a filter mechanism which could be made available for all (rules, hooks) features.

The solution could be a filter property which could look something like this (this is just the result of a very short brainstorming and has to be further improved):

routing rules:
```json
{
	"/some/fancy/url/(.*)": {
		"path": "/some/other/fancy/url/$1",
		"filter": {
			"methods": "(GET|PUT|DELETE)$",
			"headers": "regex to filter headers"
		}
	}
}
```
**some more specifications:**

- with json schema configuration, it has to be prevented to define (the exisiting) _methods_ and (new) _filter_ properties concurrently
- filter values (methods, headers) are always combined with an **AND** operation. So a rule is applied when the url matches _and_ the http methods match _and_ a header value is present

**Comment by lbovet** [comment]
The general idea looks good. I am wondering about the logical combination (AND OR NOT) we may want when defining such filters. We should keep things generic but not too complex...

**Comment by mcweba** [comment]
Released in [v1.2.7](https://github.com/swisspush/gateleen/releases/tag/v1.2.7)
